src/Components/Component.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+#include <thread>
+#include <atomic>
+#include <mutex>
+#include <queue>
+#include <iostream>
+using namespace std;
```

Comment:
Please do not use using namespace std. See: https://google.github.io/styleguide/cppguide.html#Namespaces

You should either reference your types by their full path, i.e. std::string
Or, you can add using directive in the class, not at the global scope.